processor/tailsamplingprocessor/internal/decisioncache/types.go

```diff
@@ -0,0 +1,14 @@
+package decisioncache
```

Comment:
Further idea, perhaps for a follow-up PR: make the caches observable. We probably want to know the following metrics:

- cache hits
- cache misses
- cache size

This could either be a `type CacheMetrics struct`, make the cache interfaced return those counters/gauge, or make the individual cache implementations accept the TelemetryBuilder instance we create in the new processor function.

Reply:
All good ideas, working on them now

Reply:
I think the interface suggestions make sense, but after playing around with the code I think it's easier to use just `[V any]` as the only generic, since the trace ID will always be the key, and we can make optimisations around that in the implementation layer if it's structured like this (like using an uint64 instead of pcommon.TraceID internally). This would work for replacing idToTrace as well. 

I like the idea of the observable caches as well. Would be happy to do a follow up PR for that.